doc/source/whatsnew/v1.1.0.rst

```diff
@@ -1169,6 +1169,7 @@ Other
 - Bug in :class:`Tick` comparisons raising ``TypeError`` when comparing against timedelta-like objects (:issue:`34088`)
 - Bug in :class:`Tick` multiplication raising ``TypeError`` when multiplying by a float (:issue:`34486`)
 - Passing a `set` as `names` argument to :func:`pandas.read_csv`, :func:`pandas.read_table`, or :func:`pandas.read_fwf` will raise ``ValueError: Names should be an ordered collection.`` (:issue:`34946`)
+- Improved error message for invalid construction :func:`pandas.core.internals.construction.sanitize_index`
```

Comment:
- `sanitize_index` is essentially "private" which a user shouldn't call. We'd want to reference a public API that hits this method instead (e.g. `DataFrame`, if that ends up here)
- Could you add a ```  (:issue:`PR number`) ``` at the end?